### PR TITLE
fix: set ulimit by default in startup strategy

### PR DESCRIPTION
**Pull request by feddevanderlist**
fix https://github.com/jarlah/testcontainers-ceph/issues/176 and set ulimit so startup time is reduced on platforms that support it.

This has been built and tested on an pc running manjaro.

### CODE REVIEW

**Review by jarlah — COMMENTED** [review]
Awesome! Super thanks for the contribution 🙌 i will look at this more later.

**Comment by feddevanderlist** [comment]
@jarlah I made a mistake in my thinking fixed it now. It's building again and more resilient. I might make a new merge request tomorrow that cleans up the code a little bit if you're ok with it :)

**Comment by jarlah** [comment]
Thats ok. Just tell me when you feel the PR is ready

**Comment by feddevanderlist** [comment]
it is ready now. Just saw some things in the code that I might wanna clean up. But that should not be part of this. Because I didn't make or change it. And it's your project :)

**Comment by jarlah** [comment]
> it is ready now. Just saw some things in the code that I might wanna clean up. But that should not be part of this. Because I didn't make or change it. And it's your project :)

OUR project, anyone who contribute to this repo can call it OUR repo :) im just a facilitator :bow:

File: src/main/java/org/testcontainers/containers/CephContainer.java [hunk 1]
```diff
@@ -39,6 +42,8 @@ public class CephContainer extends GenericContainer<CephContainer> {
 
     private static final String CEPH_END_START_REGEX_FORMAT = ".*Bucket 's3://%s/' created\n.*";
 
+    private static final Ulimit[] DEFAULT_ULIMITS = new Ulimit[]{new Ulimit("nofile", 65536L, 65536L)};
```
Comment (jarlah) [inline src/main/java/org/testcontainers/containers/CephContainer.java:45]: can you explain this to me. What this ulimit stuff means in layman terms. I think i know, i just want to confirm we are on the same page.

File: src/main/java/org/testcontainers/containers/CephContainer.java [hunk 2]
```diff
@@ -56,6 +61,10 @@ public CephContainer(final String dockerImageName) {
     public CephContainer(final DockerImageName dockerImageName) {
         super(dockerImageName);
         dockerImageName.assertCompatibleWith(DEFAULT_IMAGE_NAME);
+        withCreateContainerCmdModifier(createContainerCmd ->
```
Comment (jarlah) [inline src/main/java/org/testcontainers/containers/CephContainer.java:64]: i cant really see from the code what you said was the solution in the issue 

"Solution:
Set the wait strategy in the constructor of your container. Instead of in the configure. This way setting a strategy after calling new CephContainer or Super() will override the waitingStrategy."

can you elaborate how the solution matches the code. Because you are not setting wait strategy in constructor now. You are setting cmd modifier with ulimit settings

**Review by jarlah — COMMENTED** [review]
some questions

File: src/main/java/org/testcontainers/containers/CephContainer.java (hunk 2 above)
Comment (feddevanderlist) [reply src/main/java/org/testcontainers/containers/CephContainer.java:64]: I thought solving this in the constructor would be possible. Unfortunately with the variable log check that's not possible (because in the constructor ` this.cephBucket` isn't defined yet)
I have solved this by check what the waiting strategy is set in configure. (i.e. when `start()` is called)
If Strategy is default we set the logical strategy which is by logging. 
Otherwise we leave it as it's set by the user.

**Review by jarlah — APPROVED** [review]
the PR looks good. It introduces new tests and fixes existing tests. The changes are understandable. Lets merge this. I will check if it builds and run the tests locally on rootless docker on nixos. WIll release if everything works
